apps/server/src/shared/domain/entity/task-card.entity.ts

```diff
@@ -67,6 +75,36 @@ export class TaskCard extends BaseEntityWithTimestamps implements ICard, ITaskCa
 		return this.visibleAtDate < this.dueDate;
 	}
 
-	@OneToOne({ type: 'Task', fieldName: 'taskId', eager: true, unique: true, cascade: [Cascade.ALL] })
-	task!: Task;
+	public completeForUser(user: User): void {
+		this.completed.add(user);
+	}
+
+	public undoForUser(user: User): void {
+		this.completed.remove(user);
+	}
+
+	/* public isCompletedForUser(user: User): boolean {
+		const completedUserIds = this.getCompletedUserIds();
+		const isCompleted = completedUserIds.some((id) => id === user.id);
+		return isCompleted;
+	} */
+
+	public getCompletedUserIds(): string[] {
```

Comment:
Do not rely on 0 to be falsy. Alway check for boolean values.
People coming from different languages will have problems here.

Also this could be replaced with 
```js
const completedUsers =  this.completed.getItems().map((user) => user.id);

return completedUsers;
```